Core/Loyc.Essentials/Utilities/ParseHelpers.cs

```diff
@@ -390,6 +391,48 @@ static bool TryParseUInt(ref UString s, ref ulong result, int radix, ParseNumber
 			return !overflow && numDigits > 0;
 		}
 
+		public static bool TryParseUInt(ref UString s, out BigInteger result, int radix = 10, ParseNumberFlag flags = 0)
+		{
+			result = 0;
+			int _;
+			return TryParseUInt(ref s, ref result, radix, flags, out _);
+		}
+		static bool TryParseUInt(ref UString s, ref BigInteger result, int radix, ParseNumberFlag flags, out int numDigits)
+		{
+			numDigits = 0;
+			if ((flags & ParseNumberFlag.SkipSpacesInFront) != 0)
+				s = SkipSpaces(s);
+
+			int oldStart = s.InternalStart;
+
+			for (;; s = s.Slice(1))
+			{
+				char c = s[0, '\0'];
+				uint digit = (uint)Base36DigitValue(c);
+				if (digit >= radix) {
+					if ((c == ' ' || c == '\t') && (flags & ParseNumberFlag.SkipSpacesInsideNumber) != 0)
+						continue;
+					else if (c == '_' && (flags & ParseNumberFlag.SkipUnderscores) != 0)
+						continue;
+					else if (c == '\'' && (flags & ParseNumberFlag.SkipSingleQuotes) != 0)
+						continue;
+					else
+						break;
+				}
+
+				BigInteger next;
+				try {
+					next = result * (uint)radix + digit;
+				} catch (OutOfMemoryException) {
```

Comment:
I don't think OutOfMemoryException should be caught, any more than we would treat a file that is too long as a syntax error. If a number is crazy long, the symptom noticed is far more likely to be slow performance (the parser is O(N^2)).